build.gradle

```diff
@@ -1,56 +1,58 @@
-    compile "io.grpc:grpc-core:1.13.1"
-    compile "io.grpc:grpc-netty-shaded:1.13.1"
-    compile "io.grpc:grpc-protobuf:1.13.1"
-    compile "io.grpc:grpc-protobuf-lite:1.13.1"
-    compile "io.grpc:grpc-stub:1.13.1"
-    compile "io.opencensus:opencensus-api:0.12.3"
-    compile "io.opencensus:opencensus-contrib-grpc-metrics:0.12.3"
-    compile "joda-time:joda-time:2.9.2"
-    compile "org.apache.httpcomponents:httpclient:4.0.1"
-    compile "org.apache.httpcomponents:httpcore:4.0.1"
-    compile "org.threeten:threetenbp:1.3.3"
+    implementation "com.fasterxml.jackson.core:jackson-core:2.1.3" // is it needed?
+    implementation "com.google.api:api-common:2.1.2"
+    implementation "com.google.api:gax:2.8.1"
+    implementation "com.google.api:gax-grpc:2.8.1"
+    implementation "com.google.api.grpc:grpc-google-cloud-pubsub-v1:1.19.1" // is it needed?
+    implementation "com.google.api.grpc:proto-google-cloud-pubsub-v1:1.97.1"
+    implementation "com.google.api.grpc:proto-google-common-protos:2.7.1"
+    implementation "com.google.api.grpc:proto-google-iam-v1:1.2.0"
+    implementation "com.google.auth:google-auth-library-credentials:1.3.0"
+    implementation "com.google.auth:google-auth-library-oauth2-http:1.3.0"
+    implementation "com.google.auto.value:auto-value:1.4" // is it needed ?
+    implementation "com.google.cloud:google-cloud-core:1.37.1" // is it needed?
+    implementation "com.google.cloud:google-cloud-core-grpc:1.37.1" // is it needed?
+    implementation "com.google.code.findbugs:jsr305:3.0.2"
+    implementation "com.google.code.gson:gson:2.8.9"
+    implementation "com.google.errorprone:error_prone_annotations:2.10.0"
+    implementation "com.google.guava:guava:31.0.1-jre"
+    implementation "com.google.http-client:google-http-client:1.41.0"
+    implementation "com.google.http-client:google-http-client-jackson2:1.19.0" // is it needed?
+    implementation "com.google.protobuf:protobuf-java:3.19.2"
+    implementation "com.google.protobuf:protobuf-java-util:3.19.2"
+    implementation "com.google.protobuf:protobuf-lite:3.0.1" // is it needed?
+    implementation "commons-codec:commons-codec:1.15"
+    implementation "commons-logging:commons-logging:1.2"
+    implementation "io.grpc:grpc-auth:1.43.2"
+    implementation "io.grpc:grpc-context:1.43.2"
+    implementation "io.grpc:grpc-core:1.43.2"
+    implementation "io.grpc:grpc-netty-shaded:1.43.2"
+    implementation "io.grpc:grpc-protobuf:1.43.2"
+    implementation "io.grpc:grpc-protobuf-lite:1.43.2"
+    implementation "io.grpc:grpc-stub:1.43.2"
+    implementation "io.opencensus:opencensus-api:0.28.0"
+    implementation "io.opencensus:opencensus-contrib-grpc-metrics:0.12.3" // is it needed?
+    implementation "joda-time:joda-time:2.9.2" // is it needed?
+    implementation "org.apache.httpcomponents:httpclient:4.5.13"
+    implementation "org.apache.httpcomponents:httpcore:4.4.15"
+    implementation "org.threeten:threetenbp:1.5.2"
```

Comment:
Unfortunately, the change isn't quite as simple as updating the dependency list - the dependency list looks to have changed quite significantly since the last time this plugin was built - eg `gson` is preferred to `jackson`, which led to a  
```
java.lang.NoClassDefFoundError: com/google/api/client/json/gson/GsonFactory
```
error when trying to run the plugin against a pub/sub account. 

Running a dependency check on the google pubsub dependency, I got:

```
runtimeClasspath - Runtime classpath of source set 'main'.
\--- com.google.cloud:google-cloud-pubsub:1.115.1
     +--- io.grpc:grpc-api:1.43.2
     +--- io.grpc:grpc-context:1.43.2
     +--- io.grpc:grpc-stub:1.43.2
     +--- io.grpc:grpc-protobuf:1.43.2
     +--- io.grpc:grpc-protobuf-lite:1.43.2
     +--- com.google.api:api-common:2.1.2
     +--- javax.annotation:javax.annotation-api:1.3.2
     +--- com.google.protobuf:protobuf-java:3.19.2
     +--- com.google.api.grpc:proto-google-common-protos:2.7.1
     +--- com.google.auth:google-auth-library-oauth2-http:1.3.0
     +--- com.google.auth:google-auth-library-credentials:1.3.0
     +--- com.google.http-client:google-http-client-gson:1.41.0
     +--- com.google.code.gson:gson:2.8.9
     +--- com.google.api.grpc:proto-google-cloud-pubsub-v1:1.97.1
     |    +--- javax.annotation:javax.annotation-api:1.3.2
     |    +--- com.google.auto.value:auto-value-annotations:1.9
     |    +--- com.google.guava:failureaccess:1.0.1
     |    +--- com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava
     |    +--- com.google.code.findbugs:jsr305:3.0.2
     |    +--- org.checkerframework:checker-qual:3.20.0
     |    +--- com.google.errorprone:error_prone_annotations:2.10.0
     |    \--- com.google.j2objc:j2objc-annotations:1.3
     +--- com.google.api.grpc:proto-google-iam-v1:1.2.0
     +--- com.google.guava:guava:31.0.1-jre
     +--- com.google.guava:failureaccess:1.0.1
     +--- com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava
     +--- org.checkerframework:checker-qual:3.20.0
     +--- com.google.j2objc:j2objc-annotations:1.3
     +--- com.google.api:gax:2.8.1
     +--- com.google.api:gax-grpc:2.8.1
     +--- io.grpc:grpc-alts:1.43.2
     +--- io.grpc:grpc-grpclb:1.43.2
     +--- org.conscrypt:conscrypt-openjdk-uber:2.5.1
     +--- io.grpc:grpc-auth:1.43.2
     +--- io.grpc:grpc-netty-shaded:1.43.2
     +--- io.grpc:grpc-xds:1.43.2
     +--- io.grpc:grpc-services:1.43.2
     +--- com.google.re2j:re2j:1.5
     +--- org.bouncycastle:bcpkix-jdk15on:1.67
     +--- org.bouncycastle:bcprov-jdk15on:1.67
     +--- io.opencensus:opencensus-proto:0.2.0
     +--- com.google.protobuf:protobuf-java-util:3.19.2
     +--- org.threeten:threetenbp:1.5.2
     +--- com.google.code.findbugs:jsr305:3.0.2
     +--- io.opencensus:opencensus-api:0.28.0
     +--- io.grpc:grpc-core:1.43.2
     +--- com.google.android:annotations:4.1.1.4
     +--- org.codehaus.mojo:animal-sniffer-annotations:1.20
     +--- io.perfmark:perfmark-api:0.23.0
     +--- com.google.errorprone:error_prone_annotations:2.10.0
     +--- com.google.auto.value:auto-value-annotations:1.9
     +--- com.google.http-client:google-http-client:1.41.0
     +--- org.apache.httpcomponents:httpclient:4.5.13
     +--- commons-logging:commons-logging:1.2
     +--- commons-codec:commons-codec:1.15
     +--- org.apache.httpcomponents:httpcore:4.4.15
     \--- io.opencensus:opencensus-contrib-http-util:0.28.0
```

Reply:
I've added in PR for [output bigquery](https://github.com/logstash-plugins/logstash-output-google_bigquery/pull/67/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R43).

Do you think we have to list the runtime dependencies from Gradle and once flattened the tree of transitive dependencies, list all in the here so we don't incur in a such problems? (the same has to be done for the other google plugins)

Reply:
Yes, I believe so